### PR TITLE
Livepool llm rebase

### PR DESCRIPTION
**Pull request by ad-astra-video**
**What does this pull request do? Explain your changes. (required)**

Rebases [Livepool LLM PR](https://github.com/livepeer/go-livepeer/pull/3114)

**Specific updates (required)**

- rebased Livepool LLM PR for codegen updates and couple other minor updats

**How did you test each of these updates (required)**

- modified reflex-chat repo to test chat completions with history using Livepool openai-middleware 
- curl cmd sent to gateway with `stream=false` and `stream=true`
  - `curl -N -X POST http://127.0.0.1:6666/llm-generate -F "prompt=tell me a story about Mario taking a beach vacation" -F "model_id=meta-llama/Meta-Llama-3.1-8B-Instruct" -F max_tokens=256 -F stream=true`

**Does this pull request close any open issues?**

**Checklist:**

- [ ] Read the [contribution guide](./CONTRIBUTING.md)
- [X] `make` runs successfully
- [ ] All tests in `./test.sh` pass
- [ ] README and other documentation updated
- [ ] [Pending changelog](./CHANGELOG_PENDING.md) updated

### CODE REVIEW

File: core/ai.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ type AI interface {
 	Upscale(context.Context, worker.GenUpscaleMultipartRequestBody) (*worker.ImageResponse, error)
 	AudioToText(context.Context, worker.GenAudioToTextMultipartRequestBody) (*worker.TextResponse, error)
 	SegmentAnything2(context.Context, worker.GenSegmentAnything2MultipartRequestBody) (*worker.MasksResponse, error)
+	LlmGenerate(context.Context, worker.GenLlmFormdataRequestBody) (interface{}, error)
```
Comment (rickstaa) [inline core/ai.go:26]: @kyriediculous, @ad-astra-video can we rename this to LLMGenerate?

File: server/ai_process.go [hunk 2]
```diff
@@ -792,6 +793,164 @@ func submitAudioToText(ctx context.Context, params aiRequestParams, sess *AISess
...
+	go func() {
+		defer close(streamChan)
+		defer body.Close()
+		scanner := bufio.NewScanner(body)
+		var totalTokens int
+		for scanner.Scan() {
+			line := scanner.Text()
+			if strings.HasPrefix(line, "data: ") {
+				data := strings.TrimPrefix(line, "data: ")
+				if data == "[DONE]" {
```
Comment (rickstaa) [inline server/ai_process.go:890]: @kyriediculous why do we need both status `DONE` and a done boolean?

File: core/ai.go (hunk 1 above)
Comment (kyriediculous) [reply core/ai.go:26]: Yes sounds good

File: server/ai_process.go (hunk 2 above)
Comment (ad-astra-video) [reply server/ai_process.go:890]: Per discord message, SSE streams end with a `[DONE]` message. This comment can be addressed in update to [Livepool LLM PR](https://github.com/livepeer/go-livepeer/pull/3114).

**Comment by ad-astra-video** [comment]
Closing, [Livepool LLM PR](https://github.com/livepeer/go-livepeer/pull/3114) will be used to take this across the finish line.  This PR was created to hopefully save some time on the updates needed to get there since I had to make these changes to test the pipeline end-to-end.

**Pull request closed without merging**